src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/ResourceDataNode.cs

```diff
@@ -0,0 +1,191 @@
+                blob,
+                Array.Empty<Relocation>(),
+                1,
+                new ISymbolNode[]
+                {
+                                this,
+                                EndSymbol
+                });
+        }
+
+        public List<ResourceIndexData> GetOrCreateIndexData(NodeFactory factory)
+        {
+            if (IndexData != null)
+            {
+                return IndexData;
+            }
+
+            TotalLength = 0;
+            IndexData = new List<ResourceIndexData>();
+            // Build up index information
+            foreach (EcmaAssembly module in factory.MetadataManager.GetModulesWithMetadata().OfType<EcmaAssembly>())
+            {
+                PEMemoryBlock resourceDirectory = module.PEReader.GetSectionData(module.PEReader.PEHeaders.CorHeader.ResourcesDirectory.RelativeVirtualAddress);
+
+                try
+                {
+                    checked
+                    {
+                        foreach (var resourceHandle in module.MetadataReader.ManifestResources)
+                        {
+                            ManifestResource resource = module.MetadataReader.GetManifestResource(resourceHandle);
+
+                            // Don't try to embed linked resources or resources in other assemblies
+                            if (!resource.Implementation.IsNil)
+                            {
+                                continue;
+                            }
+
+                            string resourceName = module.MetadataReader.GetString(resource.Name);
+                            string assemblyName = module.GetName().FullName;
+                            BlobReader reader = resourceDirectory.GetReader((int)resource.Offset, resourceDirectory.Length - (int)resource.Offset);
+                            int length = (int)reader.ReadUInt32();
+                            ResourceIndexData indexData = new ResourceIndexData(assemblyName, resourceName, TotalLength, (int)resource.Offset + sizeof(Int32), module, length);
+                            IndexData.Add(indexData);
+                            TotalLength += length;
+                        }
+                    }
+                }
+                catch (OverflowException)
+                {
+                    throw new BadImageFormatException();
+                }
+            }
+
+            return IndexData;
+        }
+
+        /// <summary>
+        /// Extracts resources from all modules being compiled into a single blob and saves
+        /// the information needed to create an index into that blob.
+        /// </summary>
+        private byte[] GenerateResourceBlob(NodeFactory factory)
+        {
+            GetOrCreateIndexData(factory);
+
+            // Read resources into the blob
+            byte[] resourceBlob = new byte[TotalLength];
+            int currentPos = 0;
+            foreach (ResourceIndexData indexData in IndexData)
```

Comment:
Ah, right. I missed that it populates the `TotalLength` field. (Didn't read `TotalLength` as an instance field because it's not prefixed by an underscore :P).